src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[DEFAULT_SIZE];
+    }
+
+    public ArrayList(int customSize) {
+        objects = new Object[customSize];
+    }
 
     @Override
     public void add(T value) {
-
+        if (objects.length - size <= 5) {
+            newSize();
+        }
+        objects[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (isPresentInArray(index)) {
+            for (int i = size + 1; i > index; i--) {
```

Comment:
you can use here 
System.arraycopy(objects, index, objects, index + 1, size - index);